web/__init__.py

```diff
@@ -219,6 +219,28 @@ def app_api_ankerctl_server_reload():
         return web.util.flash_redirect(url_for('app_root'), "Ankerctl reloaded successfully", "success")
 
 
+@app.post("/api/ankerctl/file/upload")
+def app_api_ankerctl_file_upload():
+    if request.method != "POST":
+        return web.util.flash_redirect(url_for('app_root'))
+    if "gcode_file" not in request.files:
+        return web.util.flash_redirect(url_for('app_root'), "No file found", "danger")
+    file = request.files["gcode_file"]
+
+    try:
```

Comment:
It becoming clear that we are doing a whole lot of manual try-except handling for flask endpoints, and it's only getting worse.

This is not a criticism of your PR, but it's something we need to solve. I'll look into refactoring our handlers, so we can catch exceptions in a more generic way, and return reasonable http messages more automatically :)